File: src/main/java/thriftyjava/example/thrifty/Entity/Product.java
package thriftyjava.example.thrifty.Entity;

import lombok.Data;
import lombok.Generated;

import javax.persistence.*;

@Entity
@Data
@Table(name = "product")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    @Column(name = "price")
    private String price;

    @Column(name = "image_url")
    private String imageUrl;

    @Column(name = "category")
    private String category;
}

File: lib/Product/Product.dart
class Product {
  final int id;
  final String title;
  final String description;
  final String price;
  final String imageUrl;
  final String category;

  Product({
    this.id,
    this.title,
    this.description,
    this.price,
    this.imageUrl,
    this.category
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      title: json['title'],
      price: json['price'],
      description: json['description'],
      imageUrl: json['image'],
      category: json['category']
    );
  }
}
